helios-client/src/main/java/com/spotify/helios/client/DefaultRequestDispatcher.java

```diff
@@ -233,9 +238,13 @@ private HttpURLConnection connect0(final URI ipUri, final String method, final b
     // We verify the TLS certificate against the original hostname since verifying against the
     // IP address will fail
     if (urlConnection instanceof HttpsURLConnection) {
+      checkNotNull(user, "username must be provided for HTTPS client auth");
```

Comment:
Maybe disable the use of client certs if `user` is `null`? It's conceivable that someone wants to use SSL without using client certificates for authentication.